lib/flowDiagnostics.js

```diff
@@ -62,43 +62,56 @@ async function getDiagnostics(document) {
 			const {level, messageComponents} = message
 			if (!messageComponents.length) return
 
-			const
-				[baseMessage, ...other] = messageComponents,
-				range = baseMessage.range;
-
-			if (range == null) return;
-
-			const file = range.file
-
-			let diag = {
-				severity: level,
-				startLine: range.start.line,
-				startCol: range.start.column,
-				endLine: range.end.line,
-				endCol: range.end.column,
-				msg: '',
-			}
-
-			let details = [];
-			other.forEach(part => {
-				let partMsg = part.descr;
-				if (partMsg && partMsg !== 'null' && partMsg !== 'undefined') {
-					details.push(partMsg);
+			const messages = [];
+			const files = [];
+			const description = [];
+
+			messageComponents.forEach(message => {
+				const range = message.range;
+
+				if (message.descr && message.descr !== 'null' && message.descr !== 'undefined') {
+					description.push(message.descr);
 				}
-			});
 
-			let msg = baseMessage.descr;
-			if (details.length) {
-				msg = `${msg} (${details.join(' ')})`;
-			}
+				if (range == null) return;
 
-			diag.msg = `[flow] ${msg}`;
+				if (files.includes(range.file)) {
+					return;
+				}
 
-			if (!diags[file]) {
-				diags[file] = [];
-			}
+				let diag = {
+					file: range.file,
+					severity: level,
+					startLine: range.start.line,
+					startCol: range.start.column,
+					endLine: range.end.line,
+					endCol: range.end.column,
+					msg: '',
+				}
 
-			diags[file].push(diag);
+				files.push(range.file);
+				messages.push(diag);
+			})
+
+			messages.forEach(message => {
+				if (!diags[message.file]) {
+					diags[message.file] = [];
+				}
+
+				message.msg = description.reduce((full, descr, i, arr) => {
```

Comment:
It seems like `description` can be an empty array, which means this might throw an exception. Maybe only call reduce if arr `description` is non-empty?